provisioning/device/src/ProvisioningRegistrationSubstatus.cs

```diff
@@ -23,6 +23,15 @@ public enum ProvisioningRegistrationSubstatusType
         /// Device has been assigned to a different IoT hub and its device data was populated from the initial state stored in the enrollment.
         /// Device data was removed from the previously assigned IoT hub.
         /// </summary>
-        DeviceDataReset = 3
+        DeviceDataReset = 3,
+
+        /// <summary>
+        /// Device has been re-provisioned to a previously assigned IoT hub.
+        /// </summary>
+        /// <remarks>
+        /// For API versions prior to "2019-04-15" a substatus of <see cref="InitialAssignment"/> was returned if the device previously existed in hub.
+        /// Starting API version "2019-04-15" a substatus of ReprovisionedToInitialAssignment is returned if the device previously existed in hub.
+        /// </remarks>
+        ReprovisionedToInitialAssignment = 4,
```

Comment:
This is a breaking change from the service, no?

Reply:
Also, customers won't know what API version we use, so we should probably indicate relevant SDK version, if we were to simply take this change.

Reply:
Yes, it is. They have signed off on it and have kept it behind an API version change.

Reply:
> Also, customers won't know what API version we use, so we should probably indicate relevant SDK version, if we were to simply take this change.

That's a good point. I'll add a note to update this once we are ready to put the release out (so that we can reference the latest nuget version numbers).

Reply:
> Yes, it is. They have signed off on it and have kept it behind an API version change.

Who signed off on it? API version changes don't protect customers who upgrade from 1 SDK version to the next. It is still a breaking change.